lib/controllers/forgot-password.js

```diff
@@ -27,45 +27,54 @@ module.exports = function (req, res) {
 
   res.locals.status = req.query.status;
 
-  if (req.method === 'POST' && accepts === 'json') {
-    return application.sendPasswordResetEmail(req.body.email, function (err) {
-      if (err) {
-        logger.info('A user tried to reset their password, but supplied an invalid email address: ' + req.body.email + '.');
+  helpers.handleAcceptRequest(req, res, {
+    // Handle incoming requests from an API-like clients (something like
+    // Angular / React / REST).
+    'application/json': function () {
+      if (req.method !== 'POST') {
+        return next();
       }
-      res.end();
-    });
-  }
 
-  if (config.web.spaRoot) {
-    return res.sendFile(config.web.spaRoot);
-  }
-
-  forms.forgotPasswordForm.handle(req, {
-    // If we get here, it means the user is submitting a password reset
-    // request, so we should attempt to send the user a password reset email.
-    success: function (form) {
-      application.sendPasswordResetEmail(form.data.email, function (err) {
+      return application.sendPasswordResetEmail(req.body.email, function (err) {
         if (err) {
-          logger.info('A user tried to reset their password, but supplied an invalid email address: ' + form.data.email + '.');
+          logger.info('A user tried to reset their password, but supplied an invalid email address: ' + req.body.email + '.');
         }
 
-        res.redirect(config.web.forgotPassword.nextUri);
+        res.end();
       });
     },
-    // If we get here, it means the user didn't supply required form fields.
-    error: function (form) {
-      // https://github.com/caolan/forms/issues/96
-      if (req.query.status === 'invalid_sptoken') {
-        return helpers.render(req, res, view, { form: form, status: req.query.status });
-      }
 
-      var formErrors = helpers.collectFormErrors(form);
-      helpers.render(req, res, view, { form: form, formErrors: formErrors });
-    },
-    // If we get here, it means the user is doing a simple GET request, so we
-    // should just render the forgot password template.
-    empty: function (form) {
-      helpers.render(req, res, view, { form: form });
+    // Handle incoming requests from an browser-like clients (something like
+    // chrome / firefox / etc.).
+    'text/html': function () {
+      forms.forgotPasswordForm.handle(req, {
```

Comment:
Should really all request methods be handled here? I think that you only need to handle GET, maybe POST.